libsolidity/interface/ErrorReporter.cpp

```diff
@@ -80,6 +84,16 @@ void ErrorReporter::error(Error::Type _type, SourceLocation const& _location, Se
 	m_errorList.push_back(err);
 }
 
+void ErrorReporter::abortIfExcessive()
+{
+	if (m_errorList.size() > 256)
+	{
+		auto err = make_shared<Error>(Error::Type::Warning);
+		*err << errinfo_comment("There are more than 256 errors. Aborting.");
+		m_errorList.push_back(err);
+		BOOST_THROW_EXCEPTION(FatalError());
```

Comment:
In this case, we need to add a fatal error catcher into every sub-component, like SyntaxChecker.